obal/data/modules/koji_verify_tag.py

```diff
@@ -0,0 +1,70 @@
+"""
+Verify packages against a tag in Koji
+"""
+
+import os
+import subprocess
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.obal import get_specfile_name # pylint:disable=import-error,no-name-in-module
+
+def main():
+    """
+    Verify a tag
+    """
+    module = AnsibleModule(
+        argument_spec=dict(
+            packages=dict(type='dict', required=True),
+            tag=dict(type='str', required=True),
+            directory=dict(type='str', required=True)
+        )
+    )
+
+    tag_name = module.params['tag']
+
+    packages_for_tag = []
+
+    for (package, attributes) in module.params['packages'].items():
+        tag = None
+
+        if 'koji_tags' in attributes:
+            for koji_tag in attributes['koji_tags']:
+                if tag_name == koji_tag['name']:
+                    tag = koji_tag
+
+            if tag and 'package_base_dir' in attributes:
+                specfile = "%s/%s/%s.spec" % (attributes['package_base_dir'], package, package)
+                scl = tag['scl'] if 'scl' in tag.keys() else None
+                name = get_specfile_name(os.path.join(module.params['directory'], specfile), scl)
+                packages_for_tag.append(name)
+
+    try:
+        command = ['koji', 'list-pkgs', '--quiet', '--tag', tag_name]
+        packages_in_koji = subprocess.check_output(
+            command,
+            stderr=subprocess.STDOUT,
+            universal_newlines=True
+        )
+
+        packages_in_koji = [item.split(' ')[0] for item in packages_in_koji.split("\n")]
+        packages_in_koji = [package for package in packages_in_koji if len(package) > 0]
+
+    except subprocess.CalledProcessError as error:
+        module.fail_json(changed=False, msg=error.output)
+
+    missing_in_koji = []
+    missing_in_git = []
+
+    for package in packages_for_tag:
+        if package not in packages_in_koji:
+            missing_in_koji.append(package)
+
+    for package in packages_in_koji:
+        if package not in packages_for_tag:
+            missing_in_git.append(package)
```

Comment:
Sets are your friend
```suggestion
    missing_in_koji = packages_for_tag - packages_in_koji
    missing_in_git = packages_in_koji - packages_for_tag
```